roxctl/generate/netpol/netpol.go

```diff
@@ -25,39 +29,60 @@ type generateNetpolCommand struct {
 func Command(cliEnvironment environment.Environment) *cobra.Command {
 	generateNetpolCmd := &generateNetpolCommand{env: cliEnvironment}
 	c := &cobra.Command{
-		Use: "netpol <folder-path>",
+		Use:  "netpol <folder-path>",
+		Args: cobra.ExactArgs(1),
 		RunE: func(c *cobra.Command, args []string) error {
-			if err := validate(args); err != nil {
+			if err := generateNetpolCmd.construct(args, c); err != nil {
 				return err
 			}
-			if err := generateNetpolCmd.construct(args); err != nil {
+			if err := generateNetpolCmd.validate(); err != nil {
 				return err
 			}
 			return generateNetpolCmd.generateNetpol()
 		},
 	}
-
-	c.Flags().StringVarP(&generateNetpolCmd.outputFolderPath, "output-dir", "d", "./policies", "path to the output directory for generated policies")
-	c.Flags().StringVarP(&generateNetpolCmd.outputFilePath, "output-file", "f", "./policies.yaml", "path to the output file for merged policies")
-	c.Flags().BoolVar(&generateNetpolCmd.mergePolicies, "merge-policies", false, "Merge all generated Network Policies into a single file. Combine with -f for target file")
-	c.Flags().BoolVar(&generateNetpolCmd.splitPolicies, "split-policies", false, "Create one file per Network Policy. Combine with -d for target folder")
-
-	c.MarkFlagsRequiredTogether("split-policies", "output-dir")
-	c.MarkFlagsRequiredTogether("merge-policies", "output-file")
+	c.Flags().BoolVar(&generateNetpolCmd.removeOutputPath, "remove", false, "remove the output path if it already exists")
+	c.Flags().StringVarP(&generateNetpolCmd.outputFolderPath, "output-dir", "d", "", "save generated policies into target folder - 1 file per policy")
+	c.Flags().StringVarP(&generateNetpolCmd.outputFilePath, "output-file", "f", "", "save and merge generated policies into a single YAML file")
+	c.MarkFlagsMutuallyExclusive("output-dir", "output-file")
 	return c
 }
 
-func validate(args []string) error {
-	if len(args) > 1 {
-		return errox.InvalidArgs.Newf("too many positional arguments (expected 1, got %d)", len(args))
-	}
-	if len(args) == 0 {
-		return errox.InvalidArgs.New("missing <folder-path> argument")
+func (cmd *generateNetpolCommand) construct(args []string, c *cobra.Command) error {
+	cmd.folderPath = args[0]
+	cmd.splitMode = c.Flags().Changed("output-dir")
+	cmd.mergeMode = c.Flags().Changed("output-file")
+	return nil
+}
+
+func (cmd *generateNetpolCommand) validate() error {
+	if cmd.splitMode {
+		if err := cmd.setupPath(cmd.outputFolderPath); err != nil {
+			return errors.Wrapf(err, "failed to set up folder path")
+		}
+	} else if cmd.mergeMode {
```

Comment:
[nit] here since `splitMode` and `mergeMode` are mutually exclusive you don't need to check the second variable.

Reply:
Replying to Luis: there is the stdout mode as well and then both of these flags would be unset.